# Only used on el6
# Found the reason why compile fails in detection of powerpc64 in centos 6
# kBuild/src/lib/kStuff/include/k/kDefs.h:356:4: error: #error "Port Me or define K_ENDIAN."
# hack for gcc < 4.6 and ppc64 only
# https://stackoverflow.com/a/40675229/778517

--- kBuild/src/lib/kStuff/include/k/kDefs.h.orig	2018-05-12 03:35:13.703342877 +0100
+++ kBuild/src/lib/kStuff/include/k/kDefs.h	2018-05-12 04:24:52.034917011 +0100
@@ -336,16 +336,19 @@
   /* use K_ARCH if possible. */
 # if K_ARCH_ENDIAN != K_ENDIAN_BI
 #  define K_ENDIAN K_ARCH_ENDIAN
-# elif defined(__BYTE_ORDER__) && defined(__ORDER_LITTLE_ENDIAN__) && defined(__ORDER_BIG_ENDIAN__)
-#  if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+# elif defined(__BYTE_ORDER__)
+#  if defined(__ORDER_LITTLE_ENDIAN__) && __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
 #  define K_ENDIAN K_ARCH_LITTLE
-#  elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+#  elif defined(__ORDER_BIG_ENDIAN__) && __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
 #  define K_ENDIAN K_ARCH_BIG
 #  else
 #   error "Port Me or define K_ENDIAN."
 #  endif
 # else
-#  error "Port Me or define K_ENDIAN."
+  /* __BYTE_ORDER__ is only available since GCC 4.6 */
+  /* el6 have GCC 4.4 + or - */
+  /* use same value that was set in previous versions of kBuild */
+# define K_ENDIAN K_ARCH_ENDIAN
 # endif
 #else
   /* validate the user defined value. */
